components/locid/src/parser/langid.rs

```diff
@@ -147,8 +147,12 @@ pub const fn parse_language_identifier_with_single_variant_from_iter(
                 region = Some(r);
                 position = ParserPosition::Variant;
             } else if let Ok(v) = subtags::Variant::from_bytes_manual_slice(t, start, end) {
+                if variant.is_some() {
+                    // We cannot handle multiple variants in a const context
+                    return Err(ParserError::InvalidSubtag);
+                }
```

Comment:
Nit, here and below: I think it's not possible for `variant` to be `Some` here because every time we set `variant` to `Some` we go to `ParserPosition::Variant`.

Use a debug assertion instead:

```rust
debug_assert!(variant.is_none());
```

Reply:
Done